api/nodes.go

```diff
@@ -224,7 +221,7 @@ func (n *Nodes) monitorDrainNode(ctx context.Context, nodeID string,
 			return
 		}
 
-		if node.Status == structs.NodeStatusDown {
+		if node.Status == NodeStatusDown {
```

Comment:
Ah! I see what you meant the other day when you said linters could help. This should have never happened, but thanks to the magic of goimports and human muscle memory it happens very easily!

Thanks for catching it!

Reply:
Oh, interesting!  How would a linter have helped here?

Reply:
In the past `api/` should *never* have imported from `nomad/structs`, so something like `go list -f {{ .Deps }} | grep nomad/structs` finding a match could work as a linter.

I think the idea now is that `api/` shouldn't import _anything_ from anywhere else in Nomad at all, so a similar linter that inspects `api/`'s deps on commit would be helpful.

Reply:
Nomad few packages (e.g. `plugins`, `api`) that represent public "stable" API, and we have some opinions about the graph of package dependencies and layout.  I proposed in a meeting formalizing it and adding a linter to enforce them.